### ISSUE TITLE
Error when selecting a type from the drop down menu

### ISSUE THREAD

**Issue by JoeyJFranklin** [ketho-wow/LargerMacroIconSelection#8]
The dropdown I'm talking about
![image](https://github.com/ketho-wow/LargerMacroIconSelection/assets/134342419/d7a050ab-b16a-4d0e-94ee-197bf98213b4)

I see it's not a part of your addon but I don't get the error if I disable your addon so I figured I'd let you know :)

```lua
4x SharedXML/SharedUIPanelTemplates.lua:2821: attempt to call method 'SetIconTypes' (a nil value)
[string "@SharedXML/SharedUIPanelTemplates.lua"]:2821: in function `SetIconFilter'
[string "@SharedXML/SharedUIPanelTemplates.lua"]:2707: in function `optionSelectedCallback'
[string "@SharedXML/SharedUIPanelTemplates.lua"]:2392: in function `SetSelectedValue'
[string "@SharedXML/SharedUIPanelTemplates.lua"]:2344: in function `func'
[string "@SharedXML/UIDropDownMenu.lua"]:1008: in function `UIDropDownMenuButton_OnClick'
[string "*UIDropDownMenuTemplates.xml:131_OnClick"]:1: in function <[string "*UIDropDownMenuTemplates.xml:131_OnClick"]:1>

Locals:
self = MacroPopupFrame {
 GetIndexOfIcon = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2834
 UpdateMacroFramePanelWidth = <function> defined @Blizzard_MacroUI/Blizzard_MacroIconSelector.lua:121
 GetIconByIndex = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2830
 Update = <function> defined @Blizzard_MacroUI/Blizzard_MacroIconSelector.lua:61
 editBoxHeaderText = "Enter Macro Name (Max 16 Characters):"
 GetIconFilter = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2826
 SearchBox = MacroPopupFrameSearchBox {
 }
 SetSelectedIconText = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2792
 ReevaluateSelectedIcon = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2846
 BG = Texture {
 }
 IconSelector = Frame {
 }
 mode = 2
 iconDataProvider = <table> {
 }
 OnShow = <function> defined @Blizzard_MacroUI/Blizzard_MacroIconSelector.lua:3
 GetNumIcons = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2838
 OnHide = <function> defined @Blizzard_MacroUI/Blizzard_MacroIconSelector.lua:37
 iconFilter = 2
 GetSelectedIndex = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2842
 OnEvent = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2739
 GetMacroFrame = <function> defined @Blizzard_MacroUI/Blizzard_MacroIconSelector.lua:117
 OkayButton_OnClick = <function> defined @Blizzard_MacroUI/Blizzard_MacroIconSelector.lua:93
 BorderBox = Frame {
 }
 SetIconFromMouse = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2759
 0 = <userdata>
 SetIconFilter = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2814
 CancelButton_OnClick = <function> defined @Blizzard_MacroUI/Blizzard_MacroIconSelector.lua:88
 OnLoad = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2678
}
iconFilter = 2
iconTypes = <table> {
 1 = 1
}
(*temporary) = nil
(*temporary) = <table> {
 GetIndexOfIcon = <function> defined @LargerMacroIconSelection/Mainline/IconDataProvider.lua:159
 SetIconData = <function> defined @LargerMacroIconSelection/Mainline/IconDataProvider.lua:182
 GetIconByIndex = <function> defined @LargerMacroIconSelection/Mainline/IconDataProvider.lua:134
 Init = <function> defined @LargerMacroIconSelection/Mainline/IconDataProvider.lua:106
 Release = <function> defined @LargerMacroIconSelection/Mainline/IconDataProvider.lua:174
 extraIcons = <table> {
 }
 GetIconForSaving = <function> defined @LargerMacroIconSelection/Mainline/IconDataProvider.lua:150
 GetNumIcons = <function> defined @LargerMacroIconSelection/Mainline/IconDataProvider.lua:128
}
(*temporary) = <table> {
 1 = 1
}
(*temporary) = "attempt to call method 'SetIconTypes' (a nil value)"
IconSelectorPopupFrame_IconFilterToIconTypes = <function> defined @SharedXML/SharedUIPanelTemplates.lua:2665

```

**Comment by Ketho** [comment]
I'll have to look at it this weekend, yay for work life 💀

**Comment by Ketho** [comment]
I found the issue with the new dropdown, but I need more time for a proper fix >.< I also need to test it on classic era

**Comment by Ketho** [comment]
Well it should be finally fixed in 1.9.5
it does work a bit weird as it will always select "Spells" in the dropdown so icons do not show up double.

https://github.com/ketho-wow/LargerMacroIconSelection/releases/tag/v1.9.5
